Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs

```diff
@@ -2194,187 +2200,230 @@ private void ActPasswordHandler(ActPassword actPassword)
         }
 
         /// <summary>
-        /// Handles the synchronization of web elements using various synchronization operations.
+        /// Retrieves the appropriate locators for the ActWebSmartSync action based on the provided parameters.
         /// </summary>
-        /// <param name="act">The ActWebSmartSync object containing the synchronization details.</param>
-        public void WebSmartSyncHandler(ActWebSmartSync act)
+        /// <param name="act">The ActWebSmartSync action containing the synchronization details.</param>
+        /// <param name="pomExecutionUtil">The POMExecutionUtils object used to retrieve information about the current POM.</param>
+        /// <returns>A tuple containing the locateBy and locateValue for the ActWebSmartSync action.</returns>
+        internal (eLocateBy locateBy, string locateValue) GetLocatorsForWebSmartSync(ActWebSmartSync act, POMExecutionUtils pomExecutionUtil)
         {
-            By elementLocator = null;
-            if (act.SyncOperations != ActWebSmartSync.eSyncOperation.AlertIsPresent && act.SyncOperations != ActWebSmartSync.eSyncOperation.PageHasBeenLoaded && act.SyncOperations != ActWebSmartSync.eSyncOperation.UrlMatches)
+            eLocateBy locateBy = act.ElementLocateBy;
+            string locateValue = act.ElementLocateValueForDriver;
+
+            if (act.ElementLocateBy == eLocateBy.POMElement)
             {
-                eLocateBy locateBy = act.ElementLocateBy;
-                string locateValue = act.ElementLocateValueForDriver;
+                if (pomExecutionUtil.GetCurrentPOM() == null)
+                {
+                    throw new Exception("Relevant POM not found");
+                }
 
-                if (act.ElementLocateBy == eLocateBy.POMElement)
+                ElementInfo currentPOMElementInfo = pomExecutionUtil.GetCurrentPOMElementInfo();
+                if (currentPOMElementInfo == null)
+                {
+                    throw new Exception("Unable to find details about the POM");
+                }
+                ElementLocator firstLocator=null;
+                if (act.UseAllLocators == true)
                 {
-                    POMExecutionUtils pomExcutionUtil = new(act, act.ElementLocateValue);
-                    if (pomExcutionUtil.GetCurrentPOM() == null)
+                    foreach (var l in currentPOMElementInfo.Locators)
```

Comment:
How is it trying for all available locators where it is returning first active supported locator from line 2234

Reply:
changed code